libs/motor_control/esc/esc.h

```diff
@@ -0,0 +1,173 @@
+/*
+ * This file is part of the efc project <https://github.com/eurus-project/efc/>.
+ * Copyright (c) 2024-2025 The efc developers.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @file   esc.h
+ * @brief  Zephyr Electronic Speed Controller ESC driver for BLDC motors
+ * @author Eurus-project
+ */
+
+#ifndef ESC_H
+#define ESC_H
+
+/********************************** Includes **********************************/
+#include <stdio.h>
+#include <zephyr/drivers/pwm.h>
+#include <zephyr/kernel.h>
+
+/*********************************** Defines **********************************/
+#define MIN_THROTTLE_OFFSET_ON 1
+
+/* NOTE: Manually applied offset bellow is the placeholder for future
+implementation of ESC parameters being stored in NVS (Non Volatile Storage).
+Minimum throttle values can vary from vendor to vendor. In future, it will be
+user-configurable according to used ESCss */
+
+#if MIN_THROTTLE_OFFSET_ON
+// ESC Protocol Duty Cycle range in us
+#define DC_MIN_PWM_US 1300
+#define DC_MAX_PWM_US 2000
+
+#define DC_MIN_ONESHOT_125_US 162.5
+#define DC_MAX_ONESHOT_125_US 250
+
+#define DC_MIN_ONESHOT_42_US 54.6
+#define DC_MAX_ONESHOT_42_US 84
+
+#define DC_MIN_MULTISHOT_US 11
+#define DC_MAX_MULTISHOT_US 25
+
+#else
+// ESC Protocol Duty Cycle range in us
+#define DC_MIN_PWM_US 1000
+#define DC_MAX_PWM_US 2000
+
+#define DC_MIN_ONESHOT_125_US 125
+#define DC_MAX_ONESHOT_125_US 250
+
+#define DC_MIN_ONESHOT_42_US 42
+#define DC_MAX_ONESHOT_42_US 84
+
+#define DC_MIN_MULTISHOT_US 5
+#define DC_MAX_MULTISHOT_US 25
+
+#endif
+
+// TODO: Check this!
+//  ESC Protocol Period in us
+#define T_PWM_US 4000
+#define T_ONESHOT_125 500
+#define T_ONESHOT_42 168
+#define T_MULTISHOT 50
+
+// ESC Arming
+#define ARM_DC_PWM_US 1000
+#define ARM_DC_ONESHOT_125_US 125
+#define ARM_DC_ONESHOT_42_US 42
+#define ARM_DC_MULTISHOT_US 5
+#define ARM_DURATION 3000 // NOTE: Arm duration is around 2000-5000 ms
+
+/*********************************** Macros ***********************************/
+
+/*********************************** Typedefs *********************************/
+typedef int status_t;
+typedef uint8_t esc_number_t;
+typedef uint32_t esc_period_t;
+typedef uint32_t esc_dc_min_t, esc_dc_max_t;
+
+/**
+ * @brief ESC Protocol type
```

Comment:
Nitpick: These comments seem pretty useless considering they just repeat what is already spelled out in the type name.